pxtlib/emitter/emitter.ts

```diff
@@ -448,10 +455,76 @@ namespace ts.pxt {
-            return userError(9202, lf("Unsupported syntax node: {0}", stringKind(n)) + addInfo);
+        function unhandled(n: Node, info?: string) {
+            //If we info then we may as well present that instead
+            if (info) {
+                return userError(9202, lf("Sorry, this language feature isn't supported (${0})", info))
+            }
+
+            if (!n) {
+                console.log(`Error: ${getName(n)} is not a supported syntax feature`)
+                userError(9202, lf("Sorry, this language feature isn't supported"))
+            }
+
+            let syntax = stringKind(n)
+            let maybeSupportInFuture = false
+            let alternative: string = null
+            switch (n.kind) {
+                case ts.SyntaxKind.ForInStatement:
+                    syntax = "for in loops"
+                    break
+                case ts.SyntaxKind.ForOfStatement:
+                    syntax = "for of loops"
+                    maybeSupportInFuture = true
+                    break
+                case ts.SyntaxKind.PropertyAccessExpression:
+                    syntax = "property access"
+                    break
+                case ts.SyntaxKind.DeleteExpression:
+                    syntax = "delete"
+                    break
+                case ts.SyntaxKind.GetAccessor:
+                    syntax = "get accessor method"
+                    maybeSupportInFuture = true
+                    break
+                case ts.SyntaxKind.SetAccessor:
+                    syntax = "set accessor method"
+                    maybeSupportInFuture = true
+                    break
+                case ts.SyntaxKind.TaggedTemplateExpression:
+                    syntax = "tagged templates"
+                    break
+                case ts.SyntaxKind.ObjectLiteralExpression:
+                    syntax = "object literals"
+                    alternative = "define a class instead"
+                    break
+                case ts.SyntaxKind.TypeOfExpression:
+                    syntax = "typeof"
+                    break
+                case ts.SyntaxKind.SpreadElementExpression:
+                    syntax = "spread"
+                    break
+                case ts.SyntaxKind.TryStatement:
+                case ts.SyntaxKind.CatchClause:
+                case ts.SyntaxKind.FinallyKeyword:
+                case ts.SyntaxKind.ThrowStatement:
+                    syntax =  "throwing and catching exceptions"
+                    break
+                case ts.SyntaxKind.ClassExpression:
+                    syntax = "class expressions"
+                    alternative = "declare a class as class C {} not let C = class {}"
+                    break
+                default:
+                    break
+            }
+
+            let formatStr = maybeSupportInFuture ? "{0} not currently supported" : "{0} not supported"
+            let msg = lf(formatStr, lf(syntax))
```

Comment:
the utility that scans the code won't be able to pick up the strings statically. You need to apply lf to each of them.